File: java/cours/src/session1/td/Nat.java
package session1.td;

import hierarchie.SemiAnneauUnitaireEuclidien;

public interface Nat extends FabriqueNaturels<Nat>, SemiAnneauUnitaireEuclidien<Nat>{
	
	public boolean estNul();
	public Nat predecesseur();
	public int chiffre(int i);
	public int taille();
	public int val();
}

File: typescript/Nat.ts
import { SemiAnneauUnitaireEuclidien } from "./hierarchie/SemiAnneauUnitaireEuclidien";
import { FabriqueNaturels } from "./FabriqueNaturels";

export interface Nat extends FabriqueNaturels<Nat>, SemiAnneauUnitaireEuclidien<Nat>{
    estNul(): boolean;
    predecesseur(): Nat;
    chiffre(i: number): number;
    taille(): number;
    val(): number;
    equals(n: Nat): boolean;
}